### PR TITLE
Use consistent formatting for build system errors

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

These look pretty different from the help / hint messages we show on resolver failure. I've added color, backticks, and a "hint:" prefix.

Before:

Screenshot 2024-11-21 at 5 45 40 PM

After:

Screenshot 2024-11-21 at 6 05 31 PM

### CODE REVIEW

File: crates/uv-resolver/src/pubgrub/report.rs [hunk 1]
```diff
@@ -1105,7 +1105,7 @@ impl std::fmt::Display for PubGrubHint {
             Self::InvalidPackageMetadata { package, reason } => {
                 write!(
                     f,
-                    "{}{} Metadata for {} could not be parsed:\n{}",
+                    "{}{} Metadata for `{}` could not be parsed:\n{}",
                     "hint".bold().cyan(),
                     ":".bold(),
                     package.bold(),
```
Comment (charliermarsh) [inline crates/uv-resolver/src/pubgrub/report.rs:1111]: Should the package names be cyan? In the build failures, they're now cyan (but not bold)... But here, they're just bold.

File: crates/uv-resolver/src/pubgrub/report.rs [hunk 2]
```diff
@@ -1068,7 +1068,7 @@ impl std::fmt::Display for PubGrubHint {
             Self::PrereleaseRequested { package, range } => {
                 write!(
                     f,
-                    "{}{} {} was requested with a pre-release marker (e.g., {}), but pre-releases weren't enabled (try: `{}`)",
+                    "{}{} `{}` was requested with a pre-release marker (e.g., {}), but pre-releases weren't enabled (try: `{}`)",
```
Comment (charliermarsh) [inline crates/uv-resolver/src/pubgrub/report.rs:1071]: Maybe this is bad? We don't use backticks around the package name in the PubGrub error message (just this hint, now).

Comment (zanieb) [reply crates/uv-resolver/src/pubgrub/report.rs:1071]: I think that's consistent with everything else. The PubGrub message is the only place we omit the backtricks, I'm not sure what to do about that in the long-term.

File: crates/uv-resolver/src/pubgrub/report.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-resolver/src/pubgrub/report.rs:1111]: Consistency seems nice. I don't have strong opinions on coloring.

**Review by zanieb — APPROVED** [review]

**Pull request merged**